localstack-core/localstack/services/s3/provider.py

```diff
@@ -2011,7 +2012,7 @@ def restore_object(
             return RestoreObjectOutput()
 
         restore_expiration_date = add_expiration_days_to_datetime(
-            datetime.datetime.utcnow(), restore_days
+            datetime.datetime.now(datetime.UTC), restore_days
```

Comment:
`datetime.datetime.utcnow()` is now deprecated and should be replaced by `datetime.datetime.now(datetime.UTC)`